worker-workflow/src/main/java/com/github/cafdataprocessing/WorkflowWorker.java

```diff
@@ -0,0 +1,203 @@
+                dependenciesStatusBuilder.append(":");
+                dependenciesStatusBuilder.append(healthDependency.getStatus().toString());
+            }
+            healthMonitor.reportUnhealthy("Processing API communication is unhealthy. Service dependencies:"
+                    +dependenciesStatusBuilder.toString());
+        } catch (final Exception e) {
+            LOG.error("Problem encountered when contacting Processing API to check health: ", e);
+            healthMonitor.reportUnhealthy("Processing API communication is unhealthy: " + e.getMessage());
+        }
+    }
+
+    /**
+     * Retrieves a new AdminApi instance configured to talk to the currently set processing API for this worker.
+     * @return AdminApi pointing to configured processing API for the worker.
+     */
+    private AdminApi getWorkflowAdminApi() {
+        final ApiClient apiClient = new ApiClient();
+        apiClient.setBasePath(this.processingApiUrl);
+        return new AdminApi(apiClient);
+    }
+
+    private static WorkflowWorkerConfiguration getWorkflowWorkerConfiguration(final Application application) {
+        try {
+            return application
+                    .getService(ConfigurationSource.class)
+                    .getConfiguration(WorkflowWorkerConfiguration.class);
+        } catch (final ConfigurationException e) {
+            LOG.warn("Unable to load WorkflowWorkerConfiguration.");
+            return new WorkflowWorkerConfiguration();
+        }
+    }
+
+    /**
+     * Processes a single document. Retrieving the workflow it refers to, transforming that workflow to a runnable script,
+     * evaluating the document against the workflow to determine where it should be sent to and storing the workflow
+     * on the document so the next worker may re-evaluate the document once it has finished its action.
+     *
+     * @param document the document to be processed.
+     * @throws InterruptedException if any thread has interrupted the current thread
+     * @throws DocumentWorkerTransientException if the document could not be processed
+     */
+    @Override
+    public void processDocument(final Document document) throws InterruptedException, DocumentWorkerTransientException
+    {
+        // Get the worker task properties passed in custom data
+        final ExtractedProperties extractedProperties = CustomDataExtractor.extractPropertiesFromCustomData(document);
+        if(!extractedProperties.isValid()) {
+            LOG.warn("Custom data on document is not valid for this worker. Processing of this document will not " +
+                    "proceed for this worker.");
+            return;
+        }
+
+        // Get the specified workflow and transform it to a JavaScript representation
+        final TransformWorkflowResult transformWorkflowResult = transformWorkflow(extractedProperties, document);
+        if(transformWorkflowResult==null){
```

Comment:
Is it ever valid for this to be `null`?  Let's discuss this branch when you get a chance.

Reply:
Discussed. It is valid for this to be null in  certain api, transformation and datastore exceptions.